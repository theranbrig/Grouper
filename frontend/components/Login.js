import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import FormStyles from './styles/FormStyles';
import { Grid, Form, Button } from 'semantic-ui-react';
import Link from 'next/link';
import gql from 'graphql-tag';
import { CURRENT_USER_QUERY } from './User';

const LOGIN_USER_MUTATION = gql`
	mutation LOGIN_USER_MUTATION($email: String!, $password: String!) {
		signin(email: $email, password: $password) {
			id
			username
			email
		}
	}
`;

class Login extends Component {
	state = {
		email: '',
		password: ''
	};

	// Save to state form information
	saveToState = e => {
		this.setState({ [e.target.name]: e.target.value });
	};

	render() {
		return (
			<FormStyles>
				<Grid container textAlign="center">
					<Grid.Column mobile={16} tablet={12} computer={8} textAlign="center">
						<div>
							<h1>Sign In to Grouper</h1>
						</div>
						<Mutation
							mutation={LOGIN_USER_MUTATION}
							variables={this.state}
							refetchQueries={[{ query: CURRENT_USER_QUERY }]}>
							{(signin, { error, loading }) => {
								if (error) return <Error error={error} />;
								return (
									<Form
										loading={loading}
										inverted
										method="post"
										onSubmit={async e => {
											e.preventDefault();
											await signin();
											this.setState({ email: '', password: '' });
										}}>
										<Form.Group>
											<Form.Input
												width={16}
												type="email"
												name="email"
												label="User Name"
												value={this.state.email}
												onChange={this.saveToState}
												placeholder="Enter Email Address"
											/>
										</Form.Group>
										<Form.Group>
											<Form.Input
												name="password"
												width={16}
												label="Password"
												type="password"
												value={this.state.password}
												onChange={this.saveToState}
											/>
										</Form.Group>
										<Button inverted type="submit">
											Log{loading ? 'ging' : ''} In
										</Button>
									</Form>
								);
							}}
						</Mutation>
						<Link href="/join">
							<a>Not yet a member?</a>
						</Link>
					</Grid.Column>
				</Grid>
			</FormStyles>
		);
	}
}

export default Login;
