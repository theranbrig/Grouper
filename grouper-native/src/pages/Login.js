import React from 'react';
import { Text, View, Button, StyleSheet } from 'react-native';

const Login = ({ history }) => (
  <View style={styles.container}>
    <Text style={styles.paragraph}>This is the Login</Text>
    <Button title="Create Product" onPress={() => history.push('/')} />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  field: {
    width: 300,
    fontSize: 20,
    borderBottomWidth: 2,
    marginBottom: 10,
  },
  paragraph: {
    textAlign: 'center',
  },
});

export default Login;
