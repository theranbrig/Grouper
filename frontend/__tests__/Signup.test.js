import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Login from '../components/Signup';

describe('<Login/>', () => {
  it('should render the login page', () => {
    const wrapper = shallow(<Login />);
    expect(toJSON(wrapper.find('h1'))).toMatchSnapshot();
    expect(toJSON(wrapper.find('a'))).toMatchSnapshot();
  });
});
