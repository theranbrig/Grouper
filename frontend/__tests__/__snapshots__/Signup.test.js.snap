// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should render the login page 1`] = `
<h1>
  Sign Up for Grouper
</h1>
`;

exports[`<Login/> should render the login page 2`] = `
<a>
  Already a member? Sign In Here.
</a>
`;
